### PR TITLE
Standardize environment variables across environments

### PR DESCRIPTION
**Pull request by courtneymyers**
Update `process.env.NODE_ENV` to be consistent between client and server app for local development, and tweak server setup code a bit.

### CODE REVIEW

**Comment by courtneymyers** [comment]
@dpgalloway I'm holding off on updating the "Dev Deploy" GitHub Actions workflow step of trying to re-use the server app's "SERVER_URL" environment variable in the client app's build step (e.g. `REACT_APP_SERVER_URL="${SERVER_URL}" npm run build`) because I don't think the "SERVER_URL" environment variable is set in memory yet at that point in time, since it's defined in the CF `manifest.yml` file only... I could be wrong, but we may need to move that out of the file if we want access to it here (but I do like it in the file, as it's tied directly to where we're setting the path on Cloud.gov).

In the client app, I have a fallback set to "/csb" if that environment variable isn't defined, so it should work as is for now. I'm going to merge this in, so we can at least get the client app initially working on Cloud.gov.

TL;DR more to discuss here, so we can figure out the shared environment variable piece out in a separate PR.
